src/mobu/services/business/notebookrunner.py

```diff
@@ -116,22 +117,40 @@ async def startup(self) -> None:
         await super().startup()
 
     async def cleanup(self) -> None:
-        shutil.rmtree(str(self._repo_dir))
-        self._repo_dir = None
+        if self._repo_hash is not None:
+            await self._repo_manager.remove(
+                url=self.options.repo_url,
+                ref=self.options.repo_ref,
+                repo_hash=self._repo_hash,
+            )
+        if self._repo_path is not None:
+            rmtree(self._repo_path)
+        self._repo_path = None
+        self._repo_hash = None
         self._notebook_filter_results = None
 
     async def initialize(self) -> None:
         """Prepare to run the business.
 
-        * Check out the repository
+        * Get notebook repo files from the repo manager
         * Parse the in-repo config
         * Filter the notebooks
         """
-        if self._repo_dir is None:
-            self._repo_dir = Path(TemporaryDirectory(delete=False).name)
-            await self.clone_repo()
+        info = await self._repo_manager.add(
+            url=self.options.repo_url, ref=self.options.repo_ref
+        )
+        self._repo_path = Path(
+            mkdtemp(prefix=f"mobu-notebooks-{self.user.username}-")
+        )
+        self._repo_hash = info.hash
 
-        repo_config_path = self._repo_dir / GITHUB_REPO_CONFIG_PATH
+        # Each running business needs its own copy of the notebooks because a
+        # github refresh webhook may come in at any time and delete the repo
+        # manager's copy of the repo. We'll then get an error when we try to
+        # read those files.
+        copytree(src=info.path, dst=self._repo_path, dirs_exist_ok=True)
```

Comment:
Other possible ways around this:
* Just don't ever delete old clones after a refresh. This would be an intentional storage leak.
* Coordinate access to the repo cache among all of the monkeys so that the monkeys know to not try to read any more notebooks until a given repo has been refreshed. This seems difficult.